rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -296,6 +306,33 @@ public void onRemoval(
                 .build();
     }
 
+   public DataRead readData(String path, long readStart, int offset, int length)
+   {
+       DataRead dataRead = new DataRead();
+       byte[] buffer = new byte[CacheConfig.getBufferSize(conf)];
+       int nread;
+       BookKeeperFactory bookKeeperFactory = new BookKeeperFactory(this);
+       CachingNativeS3FileSystem fs = null;
+       try {
+           fs = new CachingNativeS3FileSystem(bookKeeperFactory, new Path(path), conf);
+           bufferSize = CacheConfig.getBufferSize(conf);
+           FSDataInputStream inputStream = fs.open(new Path(path), bufferSize);
+           inputStream.seek(readStart);
+           nread = inputStream.read(buffer, 0, length);
+           dataRead.data = ByteBuffer.wrap(buffer, 0, nread);
+           dataRead.sizeRead = nread;
+           if (inputStream != null) {
```

Comment:
`if (inputStream != null)` is not useful here, we have already accessed inputStream before this, if it had been null we would hit NPE earlier